src/org/labkey/test/components/domain/DomainFormPanel.java

```diff
@@ -217,6 +215,21 @@ public String getPanelTitle()
         return hasPanelTitle() ? elementCache().panelTitle.getText() : null;
     }
 
+    public String getPanelErrorText()
+    {
+        return getPanelErrorWebElement().getText();
+    }
+
+    public WebElement getPanelErrorWebElement()
+    {
+        getWrapper().waitFor(()-> BootstrapLocators.errorBanner.existsIn(getDriver()),
+                "the error alert did not appear as expected", 1000);
+
+        // It would be better to not return a raw WebElement but who knows what the future holds, different alerts
+        // may show different controls.
+        return BootstrapLocators.errorBanner.existsIn(getDriver()) ? BootstrapLocators.errorBanner.findElement(getDriver()) : null;
+    }
```

Comment:
I think this should be ok for now, but a possible trip up is that the scope of the existsIn is the Driver which, I believe, is the entire page. So if there is an error anywhere on the page this will find the first one it comes across.

A backlog item for the test team is to revisit the shared test component for panels. When we revisit that we should look more closely on how we find the error banner associated with a panel.

Reply:
Good to know. Thanks.